core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -89,11 +116,112 @@ public static FileSystem get(FileSystemContext context) {
       return BaseFileSystem.create(context);
     }
 
-    public static FileSystem get(ClientContext ctx) {
-      return get(FileSystemContext.create(ctx));
+    public static FileSystem create(ClientContext ctx) {
+      return create(FileSystemContext.create(ctx));
     }
   }
 
+  /**
+   * A cache for storing {@link FileSystem} clients. This should only be used by the Factory class.
+   */
+  class Cache {
+
+    private final Object mClientCacheLock = new Object();
```

Comment:
we use "client" as a synonym for "filesystem" in a lot of places here. I'd prefer to stick with "filesystem" for consistency and because they mean slightly different things

Reply:
I can agree with that. Will follow up.